scripts/build_surface_water_heat_potentials/build_river_water_heat_potential.py

```diff
@@ -0,0 +1,438 @@
+    spatial_aggregate = spatial_aggregate.compute()
+
+    # Explicitly delete approximator and intermediate arrays
+    del river_water_heat_approximator
+    del river_discharge, ambient_temperature
+    gc.collect()
+
+    result = {
+        "spatial aggregate": spatial_aggregate,
+    }
+
+    # Only include temporal aggregate if computed
+    if temporal_aggregate is not None:
+        result["temporal aggregate"] = temporal_aggregate
+    if generate_temporal_aggregates:
+        del temporal_aggregate
+
+    return result
+
+
+def set_dask_chunk_size(
+    n_threads: int,  # Number of threads per worker,
+    memory_mb: int,  # Memory per worker in MB
+    memory_safety_factor=MEMORY_SAFETY_FACTOR,
+    n_datasets: int = 2,  # ambient temperature and river discharge datasets
+    operation_multiplier: int = 3,  # Multiplier for operation overhead
+) -> None:
+    """
+    Set the Dask chunk size based on available memory and number of threads.
+    This function calculates the chunk size for Dask arrays to optimize memory usage
+    """
+
+    chunk_size = (
+        memory_mb * memory_safety_factor / n_threads / n_datasets / operation_multiplier
+    )
+    dask.config.set({"array.chunk-size": f"{chunk_size}MB"})
+
+
+if __name__ == "__main__":
+    if "snakemake" not in globals():
+        from _helpers import mock_snakemake
+
+        snakemake = mock_snakemake(
+            "build_river_water_heat_potential",
+            clusters="39",
+            opts="",
+            ll="vopt",
+            sector_opts="",
+            planning_horizons=2050,
+        )
+
+    # Configure logging and scenario
+    configure_logging(snakemake)
+    set_scenario_config(snakemake)
+    update_config_from_wildcards(snakemake.config, snakemake.wildcards)
+
+    # Get simulation snapshots
+    snapshots: pd.DatetimeIndex = get_snapshots(
+        snakemake.params.snapshots, snakemake.params.drop_leap_day
+    )
+
+    # Load regions and district heating areas
+    regions_onshore = gpd.read_file(snakemake.input["regions_onshore"])
+    regions_onshore.set_index("name", inplace=True)
+    regions_onshore = regions_onshore.to_crs("EPSG:4326")
+
+    dh_areas = gpd.read_file(snakemake.input["dh_areas"]).to_crs("EPSG:3035")
+    # Buffer district heating areas by specified amount
+    dh_areas["geometry"] = dh_areas.geometry.buffer(snakemake.params.dh_area_buffer)
+    dh_areas = dh_areas.to_crs("EPSG:4326")
+
+    # Configure Dask for multi-threading within operations (no distributed cluster)
+    dask.config.set(scheduler="threads")  # Use threaded scheduler
+    dask.config.set(num_workers=snakemake.threads)  # Use specified number of threads
+
+    set_dask_chunk_size(
+        n_threads=snakemake.threads, memory_mb=snakemake.resources.mem_mb
+    )
+
+    # Process regions sequentially but with multi-threaded Dask operations
+    results = []
+    for i, region_name in enumerate(regions_onshore.index, 1):
```

Comment:
this is potentially slow for high-resolution runs. Is there potential to multiprocess?